NMS-14150: Update XSD URL (#5123)
OpenNMS/opennms: docs/modules/operation/pages/provisioning/introduction.adoc

[[ga-provisioning-introduction]]
= Provisioning

> Provisioning is a mechanism that imports node and service definitions either from an external source such as DNS or HTTP or via the {page-component-title} web UI.

Reviewer: ```suggestion
Provisioning is a process that imports node and service definitions either from an external source such as DNS or HTTP or via the {page-component-title} web UI.
```

> Provisioning is a process that imports node and service definitions either from an external source such as DNS or HTTP or via the {page-component-title} web UI.

Reviewer: ```suggestion
Provisioning is a process that imports node and service definitions either from an external source, such as DNS or HTTP, or via the {page-component-title} web UI.
```

Provisiond receives requests to add managed entities (nodes, IP interfaces, SNMP interfaces, services) via three basic mechanisms:

* xref:provisioning/auto-discovery.adoc[Automatic discovery] (typically via the Discovery daemon).
* xref:provisioning/directed-discovery.adoc[Directed discovery] using an import requisition (typically via the Provisioning UI).

> * xref:development:rest/nodes.adoc[Node asset import] through the RestAPI or the https://docs.opennms.com/pris/1.2.0/index.html[provisioning integration server] (PRIS).

Reviewer 2: ```suggestion
* xref:development:rest/nodes.adoc[Node asset import] through the REST API or the https://docs.opennms.com/pris/1.2.0/index.html[provisioning integration server] (PRIS).
```

{page-component-title} uses an ICMP ping sweep to find IP addresses on the network and provision node entities.
Use <<provisioning/auto-discovery.adoc#ga-provisioning-auto-discovery-detectors, auto discovery with detectors>> in addition to the ICMP IP address ping to specify services that you need to detect.

> Use xref:reference:provisioning/handlers/introduction.adoc#import-handlers[import handlers] to further control provisioning.

Reviewer: Import handlers are for getting nodes from different types of sources, typically those that aren't already xml formatted. Policies do more to control what is imported.

Author: Hmmm... okay. Does this look like a better description, then?

```suggestion
Use xref:reference:provisioning/handlers/introduction.adoc#import-handlers[import handlers] to provision nodes from other sources, including sources that may not be in XML format.
```

Reviewer: I'm good with that wording.

* Scanned to discover node entity’s interfaces (SNMP and IP).
* Interfaces are persisted.
* Service detection of each IP interface.
* Node merging.

NOTE: Merging occurs only when two automatically discovered nodes appear to be the same node.
Nodes <<provisioning/directed-discovery.adoc#discovery-directed, discovered directly>> are not included in the node merging process.

[[discovery-directed]]
== Directed discovery

Directed discovery lets you specify what you want to provision based on an existing data source such as an in-house inventory, a standalone provisioning system, or a set of element management systems.
Using an import requisition, this mechanism directs {page-component-title} to add, update, or delete a node entity exactly as defined by the external source.
No discovery process is used for finding more interfaces or services.

IMPORTANT: An import requisition is an XML definition of node entities to be provisioned from an external source into {page-component-title}.
See the https://xmlns.opennms.org/xsd/config/model-import[requisition schema] (XSD) for more information.

> === Understanding the process

Reviewer 2: ```suggestion
=== The process
```

{page-component-title} uses an ICMP ping sweep to find IP addresses on the network and provision node entities.
Use <<provisioning/auto-discovery.adoc#ga-provisioning-auto-discovery-detectors, auto discovery with detectors>> in addition to the ICMP IP address ping to specify services that you need to detect.

> Use xref:reference:provisioning/handlers/introduction.adoc#import-handlers[import handlers] to provision nodes from other sources, including sources that may not be in XML format.

Reviewer: Should be a separate paragraph, as it's a different kind of "automatic discovery"
```suggestion

Use xref:reference:provisioning/handlers/introduction.adoc#import-handlers[import handlers] to provision nodes from other data sources such as DNS zone records and VMware.
```
